src/sentry/static/sentry/app/components/events/interfaces/spans/index.tsx

```diff
@@ -1,62 +1,170 @@
+import EventsV2 from 'app/utils/discover/eventsv2';
+import {stringifyQueryObject, QueryResults} from 'app/utils/tokenizeSearch';
+import AlertMessage from 'app/components/alertMessage';
+import {TableData} from 'app/views/eventsV2/table/types';
 
-import {SentryTransactionEvent} from './types';
+import {SentryTransactionEvent, ParsedTraceType} from './types';
+import {parseTrace, getTraceDateTimeRange} from './utils';
 import TraceView from './traceView';
 
-type PropType = {
+type Props = {
   orgId: string;
   event: SentryTransactionEvent;
   eventView: EventView;
-};
+} & ReactRouter.WithRouterProps;
 
 type State = {
+  parsedTrace: ParsedTraceType;
   searchQuery: string | undefined;
 };
 
-class SpansInterface extends React.Component<PropType, State> {
+class SpansInterface extends React.Component<Props, State> {
   static propTypes = {
     event: SentryTypes.Event.isRequired,
     orgId: PropTypes.string.isRequired,
   };
 
   state: State = {
     searchQuery: undefined,
+    parsedTrace: parseTrace(this.props.event),
   };
 
+  static getDerivedStateFromProps(props: Props, state: State): State {
+    return {
+      ...state,
+      parsedTrace: parseTrace(props.event),
+    };
+  }
+
   handleSpanFilter = (searchQuery: string) => {
     this.setState({
       searchQuery: searchQuery || undefined,
     });
   };
 
+  renderTraceErrorsAlert({
+    isLoading,
+    numOfErrors,
+  }: {
+    isLoading: boolean;
+    numOfErrors: number;
+  }) {
+    if (isLoading) {
+      return null;
+    }
+
+    if (numOfErrors === 0) {
+      return null;
+    }
+
+    const label =
+      numOfErrors > 1
+        ? t(
+            `There are %d error events associated with this transaction event.`,
+            numOfErrors
+          )
+        : t(`There is an error event associated with this transaction event.`);
+
+    return (
+      <AlertMessageContainer>
+        <AlertMessage
+          alert={{
+            id: 'id',
+            message: <span>{label}</span>,
+            type: 'error',
+          }}
+          system={false}
+          hideCloseButton
+        />
+      </AlertMessageContainer>
+    );
+  }
+
   render() {
-    const {event, orgId, eventView} = this.props;
+    const {event, orgId, eventView, location} = this.props;
+    const {parsedTrace} = this.state;
+
+    // construct discover query to fetch error events associated with this transaction
+
+    const {start, end} = getTraceDateTimeRange({
+      start: parsedTrace.traceStartTimestamp,
+      end: parsedTrace.traceEndTimestamp,
+    });
+
+    const conditions: QueryResults = {
+      query: [],
+      'event.type': ['error'],
+      trace: [parsedTrace.traceID],
+    };
+
+    if (typeof event.title === 'string') {
+      conditions.transaction = [event.title];
+    }
+
+    const traceErrorsEventView = EventView.fromSavedQuery({
+      id: undefined,
+      name: `Errors related to transaction ${parsedTrace.rootSpanID}`,
```

Comment:
If this is a throwaway event view do we need it to have a name?

Reply:
I think a name is fine here.